torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,710 @@
+                as input and returns the correlation features.
+                It must expose an ``out_channels`` attribute.
+
+            update_block (MultiLevelUpdateBlock): The update block, which contains the motion encoder, and the recurrent unit.
+                It takes as input the hidden state of its recurrent unit, the context, the correlation
+                features, and the current predicted depth. It outputs an updated hidden state
+            depth_head (nn.Module): The depth head block will convert from the hidden state into changes in depth.
+            mask_predictor (nn.Module, optional): Predicts the mask that will be used to upsample the predicted flow.
+                If ``None`` (default), the flow is upsampled using interpolation.
+            slow_fast (bool): A boolean that specify whether we should use slow-fast GRU or not. See RAFT-Stereo paper
+                on section 3.4 for more detail.
+            num_iters (int): The default number of iteration update during forward pass. Take note that this will only
+                be used if num_iters is not specified on the forward call.
+        """
+        super().__init__()
+        _log_api_usage_once(self)
+
+        self.feature_encoder = feature_encoder
+        self.context_encoder = context_encoder
+
+        self.base_downsampling_ratio = feature_encoder.base_downsampling_ratio
+        self.num_level = self.context_encoder.num_level
+        self.corr_pyramid = corr_pyramid
+        self.corr_block = corr_block
+        self.update_block = update_block
+        self.depth_head = depth_head
+        self.mask_predictor = mask_predictor
+
+        hidden_dims = self.update_block.hidden_dims
+        # Follow the original implementation to do pre convolution on the context
+        # See: https://github.com/princeton-vl/RAFT-Stereo/blob/main/core/raft_stereo.py#L32
+        self.context_convs = nn.ModuleList(
+            [nn.Conv2d(hidden_dims[i], hidden_dims[i] * 3, kernel_size=3, padding=1) for i in range(self.num_level)]
+        )
+        self.slow_fast = slow_fast
+        self.num_iters = num_iters
+
+    def forward(self, image1: Tensor, image2: Tensor) -> List[Tensor]:
+        # We dont put num_iters as input to forward (like the original implementation) in order to make this class fx traceable.
+        # User can still change the number of iteration by updating the class variable self.num_iters during inference
+        num_iters = self.num_iters
+        batch_size, _, h, w = image1.shape
+        torch._assert(
+            (h, w) == image2.shape[-2:],
+            f"input images should have the same shape, instead got ({h}, {w}) != {image2.shape[-2:]}",
+        )
+
+        torch._assert(
+            # We use addition, "+", instead of "and" operator so it can pass fx symbolic trace
+            (h % self.base_downsampling_ratio + w % self.base_downsampling_ratio == 0),
```

Comment:
Even inside `torch._assert` if we put the condition with `and` operator, the fx will still complain. Since in this case the behaviour of `+` and `and` should be exactly the same, therefore I use `+` instead.

Reply:
We no longer worry about FX. We could perhaps use something more intuitive that + but no strong opinions.